### PR TITLE
ISICO-14340: retry for redis ping added

### PR DESCRIPTION
**Pull request by sarmuru2**
Host supplier is checking all redis hosts before finding available hosts. To check redis hosts, its pinging redis hosts once. To add reliability, it has to query 3 times when there is a failure and with a time delay of 1 second

### CODE REVIEW

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java [hunk 1]
```diff
@@ -26,6 +26,32 @@ public class RedisPinger {
     private static final Logger log = LoggerFactory.getLogger(RedisPinger.class);
     private static final int CONNECTION_TIMEOUT = 3000;
 
+    private static int MAX_RETRY_COUNT = 3;
+    private static int RETRY_INTERVAL = 1000;
+
+    private Integer waitObject = Integer.valueOf(1);
+
+    public boolean pingWithRetry(Host host) {
```
Comment (fthankac) [inline redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java:34]: Add a comment for the function.

**Review by fthankac — CHANGES_REQUESTED** [review]

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java (hunk 1 above)
Comment (sarmuru2) [reply redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java:42]: done

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java [hunk 2]
```diff
@@ -26,6 +26,46 @@ public class RedisPinger {
     private static final Logger log = LoggerFactory.getLogger(RedisPinger.class);
     private static final int CONNECTION_TIMEOUT = 3000;
 
+    private static int MAX_RETRY_COUNT = 3;
+    private static int RETRY_INTERVAL = 1000;
```
Comment (charybr) [inline redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java:30]: retry interval of 1 second may be very short. How about using exponential backoff? Something like
1st attempt - wait for 2s
2nd attempt - wait for 3s
3rd attempt - wait for 4.5s.
Example - https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/backoff/ExponentialBackOff.html

**Review by charybr — APPROVED** [review]

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java (hunk 2 above)
Comment (sarmuru2) [reply redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/RedisPinger.java]: its implemented

**Pull request merged**